docs/android/testing/unit_testing.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "Android Unit Testing"
+sidebar_label: "Unit Tests"
+---
+
+🚧🚧🚧 Under Construction 🚧🚧🚧
+
+## Why Do We Perform Unit Testing?
+
+Unit testing helps you build your features with confidence and ensures that your code behaves as expected. It should be a tool to assist development, not a burden. [Test-Driven Development (TDD)](https://en.wikipedia.org/wiki/Test-driven_development) is a well-known methodology where tests are written before or alongside the actual code. This approach allows you to quickly validate your code without waiting for the entire application to run.
+
+:::info
+Don't write tests just for the sake of writing tests. Tests should either help you during development or assist future developers in maintaining the codebase.
+:::
+
+Unit tests focus on testing **your code**. Avoid testing the behavior of external libraries unless absolutely necessary. If you find yourself testing a library's behavior, consider contributing to that library instead and add the test there.
+
+:::note
+There are exceptions to this rule. Sometimes, we add tests to ensure that the behavior of a library doesn't change over time. In such cases, explicitly document the reason for the test.
+:::
+
+## Testing the Public Interface
+
+Focus on testing the **public API** of your classes rather than every single function. Writing tests for all functions, especially small ones, can lead to an overwhelming number of tests that are difficult to maintain. By concentrating on the public interface, you ensure that your tests remain relevant and resilient to internal changes.
+
+When you need to access private parts of a class for testing, consider using the [VisibleForTesting](https://developer.android.com/reference/kotlin/androidx/annotation/VisibleForTesting) annotation. This annotation allows you to expose private methods or properties for testing purposes only. The [linter](../linter) ensures that this exposure is limited to the test scope.
+
+:::note
+Avoid using `VisibleForTesting` unless absolutely necessary. It’s better to design your code in a way that doesn’t require exposing private members.
+:::
+
+## Test Frameworks and Mocking
+
+The project is configured to use [JUnit 5](https://junit.org/junit5/), which should be your primary testing framework.
```

Comment:
_should_? That kinda leaves the door open, but I am wondering for what

Reply:
There is an exception bellow :/ because of Robolectric we have to use JUnit 4.